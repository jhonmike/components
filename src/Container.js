import React from 'react'
import classnames from 'classnames'
import PropTypes from 'prop-types'

import styles from './container.css'

const Container = props => {
  const {
    className,
    fluid,
    children,
    ...otherProps
  } = props

  const classes = classnames(
    props.fluid ? styles['container-fluid'] : styles.container,
    className
  )

  return (
    <div {...otherProps} className={classes}>
      {children}
    </div>
  )
}

Container.propTypes = {
  fluid: PropTypes.bool,
  className: PropTypes.string,
  children: PropTypes.node
}

export default Container
