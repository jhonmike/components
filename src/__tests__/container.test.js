import React from 'react'
import renderer from 'react-test-renderer'
import Container from '../Container'

describe('Container Component', () => {
  it('renders component', () => {
    const tree = renderer
      .create(<Container>Container</Container>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders fluid container component', () => {
    const tree = renderer
      .create(<Container fluid>Container Fluid</Container>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
