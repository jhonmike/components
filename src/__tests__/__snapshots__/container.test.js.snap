// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Component renders component 1`] = `
<div
  className="container"
>
  Container
</div>
`;

exports[`Container Component renders fluid container component 1`] = `
<div
  className="container-fluid"
>
  Container Fluid
</div>
`;
